doc/reference/trd-hil-design.md

```diff
@@ -720,6 +723,42 @@ blocking APIs, not to never implement them.  They can and should at
 times exist, but their uses cases should be narrow and constrained as
 they are fundamentally not as reusable.
 
+Rule 11: Do not include `initialize()`
+===============================
+
+Avoid `initialize()` or `start()` or `on()` functions in HIL interfaces. While
+these types of methods are intuitive when implementing a HIL, they are often not
+intuitive for users of that HIL, as it is not often clear when `initialize()`
+should be called. For example, should `initialize()` be called once when the
+board boots? Or before every use? If there are multiple users of the peripheral
+(after a virtualization layer for example) then it isn't clear which user should
+call `initialize()`. In general, it is better to require that the implementation
+track the state of the underlying hardware and configure it as needed.
+
+See https://github.com/tock/tock/issues/1035 for more discussion.
```

Comment:
@hudson-ayers's last comment on this thread is interesting. Perhaps we should consider whether our thinking here has changed.

Reply:
I'm thinking that any setup function that takes arguments needs to be in the HIL. But any init/setup/configure function without arguments is what creates ambiguity and where putting it in the HIL pushes the burden to every user of the HIL, rather than the one implementer.

Reply:
I think the Components question changes this, plus the issue with lazy initialization. We do have lots of configure calls out there. Figuring the right way to describe the right approach given the different issues in play will probably take a little wordsmithing.